xbmc/addons/FontResource.cpp

```diff
@@ -0,0 +1,87 @@
+/*
+*      Copyright (C) 2005-2017 Team XBMC
+*      http://kodi.tv
+*
+*  This Program is free software; you can redistribute it and/or modify
+*  it under the terms of the GNU General Public License as published by
+*  the Free Software Foundation; either version 2, or (at your option)
+*  any later version.
+*
+*  This Program is distributed in the hope that it will be useful,
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+*  GNU General Public License for more details.
+*
+*  You should have received a copy of the GNU General Public License
+*  along with Kodi; see the file COPYING.  If not, see
+*  <http://www.gnu.org/licenses/>.
+*
+*/
+#include "FontResource.h"
+
+#include "AddonManager.h"
+#include "dialogs/GUIDialogKaiToast.h"
+#include "filesystem/File.h"
+#include "filesystem/SpecialProtocol.h"
+#include "guilib/GUIWindowManager.h"
+#include "messaging/ApplicationMessenger.h"
+#include "messaging/helpers/DialogHelper.h"
+#include "ServiceBroker.h"
+#include "settings/Settings.h"
+
+using namespace XFILE;
+using namespace KODI::MESSAGING;
+
+using KODI::MESSAGING::HELPERS::DialogResponse;
+
+namespace ADDON
+{
+
+std::unique_ptr<CFontResource> CFontResource::FromExtension(CAddonInfo addonInfo, const cp_extension_t* ext)
+{
+  return std::unique_ptr<CFontResource>(new CFontResource(std::move(addonInfo)));
+}
+
+bool CFontResource::IsInUse() const
+{
+  std::string skin = CServiceBroker::GetSettings().GetString(CSettings::SETTING_LOOKANDFEEL_SKIN);
+  ADDONDEPS deps = CServiceBroker::GetAddonMgr().GetDepsRecursive(skin);
+  for (const auto& it : deps)
+    if (it.first == ID())
+      return true;
+
+  return false;
+}
+
+void CFontResource::OnPostInstall(bool update, bool modal)
+{
+  if (IsInUse() || (!update && !modal &&
+    HELPERS::ShowYesNoDialogText(CVariant{Name()}, CVariant{24099}) == DialogResponse::YES))
+  {
+    CGUIDialogKaiToast *toast = g_windowManager.GetWindow<CGUIDialogKaiToast>(WINDOW_DIALOG_KAI_TOAST);
+    if (toast)
+    {
+      toast->ResetTimer();
+      toast->Close(true);
+    }
+    CApplicationMessenger::GetInstance().PostMsg(TMSG_EXECUTE_BUILT_IN, -1, -1, nullptr, "ReloadSkin");
+  }
+}
+
+bool CFontResource::HasFont(std::string& path, const std::string& file) const
+{
+#ifdef TARGET_POSIX
+  std::string result = CSpecialProtocol::TranslatePathConvertCase(Path()+"/resources/"+file);
+#else
+  std::string result = Path()+"/resources/"+file;
```

Comment:
ifdef not needed here, it is already handled in `CSpecialProtocol::TranslatePathConvertCase` function

Reply:
apparently it wasn't back in may as this code was copy-pasted ;) removed.